### PR TITLE
fix: limit response header content-type to prevent xss

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
limit response header content-type to prevent xss

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=327107&iterationID=1321&pId=0&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Fix the bug that limit response header content-type to prevent xss          |
| 🇨🇳 中文    |    修复了xss安全漏洞          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/core/legacy/services/filesvc/download.go [hunk 1]
```diff
@@ -37,6 +38,19 @@ const (
 	headerContentType        = "Content-Type"
 	headerContentDisposition = "Content-Disposition"
 	HeaderContentLength      = "Content-Length" // The Content-Length entity header indicates the size of the entity-body, in bytes, sent to the recipient.
+
+	headerContentTypePng = "image/png"
+)
+
+var (
+	// allowedContentTypes is a map of allowed file code(key) and content type(value).
+	allowedContentTypes = map[string]string{
```
Comment (sfwn) [inline internal/core/legacy/services/filesvc/download.go:47]: any doc link for these file header code? please link it.

File: internal/core/legacy/services/filesvc/download.go [hunk 2]
```diff
@@ -105,9 +121,51 @@ func (svc *FileService) DownloadFile(w io.Writer, file dao.File) (headers map[st
...
+	for k, v := range allowedContentTypes {
+		if strings.HasPrefix(strings.ToLower(fileCode), strings.ToLower(k)) {
+			return v
+		}
+	}
+
+	return contentType
+}
+
+func bytesToHexString(src []byte) string {
```
Comment (sfwn) [inline internal/core/legacy/services/filesvc/download.go:152]: any util for this?

File: internal/core/legacy/services/filesvc/download.go [hunk 3]
```diff
@@ -92,8 +106,10 @@ func (svc *FileService) DownloadFile(w io.Writer, file dao.File) (headers map[st
 		headerContentDisposition: headerValueDispositionInline(file.Ext, file.DisplayName),
 		HeaderContentLength:      strconv.FormatInt(file.ByteSize, 10),
 	}
-
-	contentType := mime.TypeByExtension(file.Ext)
+	var buf bytes.Buffer
+	tee := io.TeeReader(reader, &buf)
+	contentType := GetFileContentType(tee)
+	io.ReadAll(tee)
```
Comment (sfwn) [inline internal/core/legacy/services/filesvc/download.go:112]: No need readAll in tee, `buf` just need the first few chars.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/core/legacy/services/filesvc/download.go (hunk 1 above)
Comment (chengjoey) [reply internal/core/legacy/services/filesvc/download.go:48]: added

File: internal/core/legacy/services/filesvc/download.go (hunk 2 above)
Comment (chengjoey) [reply internal/core/legacy/services/filesvc/download.go:155]: added example xss-svg normal png

File: internal/core/legacy/services/filesvc/download.go (hunk 3 above)
Comment (chengjoey) [reply internal/core/legacy/services/filesvc/download.go]: deleted, adjusted by copy(buf), copy(reader)

**Review by sfwn — APPROVED** [review]

**Pull request merged**